### PR TITLE
React Band Client error checking and able to connect to C# Band Server

### PR DESCRIPTION
**Pull request by EncompassingResidential**
Adds more error checking to see any errors from Server if a problem on the Client side.
Added line in GUI to show URL send to Band Server.

C# Band Server had to allow CORS - Cross Origin Resource Sharing to have the React App GUI and this Band Server both use LocalHost:port options.

PostMan was able to make a call to the Band Server with locahost:portA but the React GUI had it's own locahost:portB

Fixed the Cross-Origin Resource Sharing (CORS) for the React App using localhost:port calling the Band Server who was also a locahost:port

Had to add setupProxy.js and view the Band Server commit to see it's CORS setting fix.  Need to look up how to change that for production compile.
Also figured out with Chat and StackOverflow how to deal with babel dependency deprecation warning on App startup via packate.json addition of https://github.com/babel to devDependencies

### CODE REVIEW

File: bandtree-client-gui/package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "@testing-library/user-event": "^13.5.0",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
-    "react-scripts": "5.0.1",
+    "react-scripts": "^5.0.1",
```
Comment (John-Ritz-OR) [inline bandtree-client-gui/package.json:11]: Looked up the caret ^:

In a package.json file, the ^ (caret) symbol is used to specify a range of allowed versions for a package's dependency. It is part of the version range syntax in npm and other package managers like Yarn. When you see a caret in a dependency's version specification, it has the following meaning:

    Major Version Compatibility: The caret indicates that you allow updates to the package within the same major version. Major versions are the first numbers in a package's version (e.g., for react version 18.2.0, the major version is 18).

    Specific Major Version: The caret does not allow changes to the major version itself. It only permits updates within the same major version. For example, if you specify "react": "^18.2.0", it means that any version of react starting from 18.2.0 up to, but not including, 19.0.0 is allowed.

Here's how the caret version range works in your example:

    "react": "^18.2.0": This means you allow updates to react within the 18.x.x range. So, you might get newer versions like 18.2.1, 18.3.0, etc., but it prevents upgrading to version 19.0.0 or higher, which could introduce breaking changes.

    "react-scripts": "^5.0.1": Similarly, this allows updates to react-scripts within the 5.x.x range but prevents upgrading to version 6.0.0 or higher.

Using the caret (^) is a common practice because it strikes a balance between getting bug fixes and minor updates while avoiding major version changes that might introduce breaking changes to your project.

If you want to specify an exact version without allowing any updates, you can omit the caret and specify the version like this:

json

"react": "18.2.0",

This ensures that your project always uses the exact version 18.2.0 of react, and no updates will be automatically applied.

https://dev.to/adamklein/package-lock-json-in-git-or-not-50l5

package-lock.json - in GIT or not?

tl;dr
1 - Put it in GIT
2 - Commit it every-time it changes
3 - Never delete it

Even more details if you need them were in this StackOverflow listing:
https://stackoverflow.com/questions/44206782/do-i-commit-the-package-lock-json-file-created-by-npm-5

**Review by John-Ritz-OR — APPROVED** [review]
AWESOME, got my 1st results from the React GUI to the C# Server and then to and from the Wikipedia API.   

Next will be munging on the data a little to present the search results to the user and showing possible Music bands or groups or artists shown from their search string.

Hard one will be bands like "The Who"  vague term and the band's Wikipedia page doesn't have "(band)" in the JSON item "Title" like many other bands on Wikipedia.
In the JSON return data, there is a "snippet" section which for the band "The Who" there is the word "band" as:
English rock <span class=\"searchmatch\">band</span> formed

Then two other important search results for bands that will be in future versions, maybe:
"List of the Who band members"
and "The Who discography"

Noticed on www for en.wikipedia.org when I searched for string "The Who" it automatically went to the correct page for the band.  It was the 1st result in the search options.

other future possibly for connections between bands:
producers, partners, children, Labels (Virgin, Warner Bros, Atlantic, etc.)
